client/my-sites/post-type-filter/index.jsx

```diff
@@ -75,70 +72,103 @@ const PostTypeFilter = React.createClass( {
 			}
 
 			return memo.concat( {
-				count: jetpack ? null : count,
+				// Hide count in all sites mode; and in Jetpack mode for non-posts
 				key: `filter-${ status }`,
-				path: [
-					'/types',
-					query.type,
+				count: ( ! siteId || ( jetpack && query.type !== 'post' ) ) ? null : count,
+				path: compact( [
+					query.type === 'post'
+						? '/posts'
+						: '/types/' + query.type,
+					query.type === 'post' && query.author && 'my',
 					pathStatus,
 					siteSlug
-				].filter( Boolean ).join( '/' ),
+				] ).join( '/' ),
 				selected: mapPostStatus( pathStatus ) === query.status,
 				children: label
 			} );
 		}, [] );
 	},
 
 	render() {
-		const { siteId, query, jetpack } = this.props;
+		const { authorToggleHidden, jetpack, query, siteId, statusSlug } = this.props;
 		const navItems = this.getNavItems();
 		const selectedItem = find( navItems, 'selected' ) || {};
 
+		const scopes = {
+			me: this.translate( 'Me', { context: 'Filter label for posts list' } ),
+			everyone: this.translate( 'Everyone', { context: 'Filter label for posts list' } )
+		};
+
 		return (
 			<div>
-				{ query && siteId && false === jetpack && (
+				{ siteId && false === jetpack && (
 					<QueryPostCounts
 						siteId={ siteId }
 						type={ query.type } />
 				) }
 				<SectionNav
-					selectedText={ selectedItem.children }
+					selectedText={
+						<span>
+							<span>{ selectedItem.children }</span>
+							{ ! authorToggleHidden && <small>{ query.author ? scopes.me : scopes.everyone }</small> }
+						</span>
+					}
 					selectedCount={ selectedItem.count }>
-					{ query && [
-						<NavTabs
-							key="tabs"
-							label={ this.translate( 'Status', { context: 'Filter group label for tabs' } ) }
-							selectedText={ selectedItem.children }
-							selectedCount={ selectedItem.count }>
-							{ navItems.map( ( props ) => <NavItem { ...props } /> ) }
-						</NavTabs>,
-						<Search
-							key="search"
-							pinned
-							fitsContainer
-							onSearch={ this.doSearch }
-							placeholder={ this.translate( 'Search…' ) }
-							delaySearch={ true } />
-					] }
+					<NavTabs
+						key="tabs"
```

Comment:
`key` should be safe to remove as we're not using an array now.

Reply:
Good spot.